services/pong/src/service.js

```diff
@@ -133,12 +136,21 @@ export default class Service {
     const session = ping.createSession();
     let online;
     let ms;
+    let ip;
 
     try {
+      if (ipRegex({exact: true, includeBoundaries: true}).test(host)) {
+        ip = host;
+      } else {
+        ip = (await promisify(dns.lookup)(host)).address;
+      }
       ms = await new Promise((resolve, reject) => {
-        session.pingHost(host, (err, _target, sent, rcvd) => {
+        let start = Date.now();
+        // Sent is undefined sometimes for unknown reasons,
+        // so start is used instead.
+        session.pingHost(ip, (err, _target, sent, rcvd) => {
           if (err) reject(err);
-          else resolve(rcvd - sent);
+          else resolve(rcvd - (sent || start));
```

Comment:
This is, for all intents and purposes, a hack. The bug is definitely in net-ping's side and needs investigation.

Reply:
yeah that should definitely be looked into, as a quick fix this works for now.